### PR TITLE
Mad/allow payload for post requests

### PR DESCRIPTION
**Pull request by CrookedGrin**
## Ticket
[EOYT: Services for Address Validation](https://www.notion.so/teamshares/EOYT-Services-for-Address-Validation-44bf79f0225c4b1db70da2b1217ce3e7?pvs=4)

## Description
The generic HTTP client did not previously allow for passing an actual payload in a POST request. This fixes that. Note that the second parameter passed to `connection.send` should always be `nil` as opposed to `payload`, which is actually the 3rd parameter. This was misleading as written previously.

## Release Reminder (check before submitting PR)
- [] I remember this is a _public_ repository (don't commit anything confidential)
- [] I'm aware I'll need to push PRs to any consuming apps that need to _use_ these changes (`bundle update --conservative teamshares_rails`)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -51,8 +51,16 @@ Instructions on testing changes to this shared engine _within another full Rails
   # e.g. bundle config --local local.teamshares_rails ../teamshares-rails
   ```
 
+
 3. Now re-run `bundle` and restart your rails server, if it was already running (note: not _technically_ sure either is necessary).
 
+4. If you're on a branch of `teamshares-rails` other than `main`, you will see an error in the console that your Gemfile is expecting to install from `main`. You can change that in your local `Gemfile` so that it points to your branch:
```
Comment (nilesmc) [inline README.md:57]: Helpful 👍🏻

**Review by nilesmc — APPROVED** [review]
Boom, love the expansion of functionality along w maintaining tight simple statements.

**Review by kdonovan — CHANGES_REQUESTED** [review]
Unfortunately [the faraday docs](https://lostisland.github.io/faraday/#/getting-started/quick-start?id=get-head-delete-trace) imply POST requests send data as the _second_ argument (the third is for headers), so we may need to bifurcate the calls rather than just `send` GETs and POSTs identically. 🤔  (@nilesmc , thoughts?)

**Comment by CrookedGrin** [comment]
> Unfortunately [the faraday docs](https://lostisland.github.io/faraday/#/getting-started/quick-start?id=get-head-delete-trace) imply POST requests send data as the _second_ argument (the third is for headers), so we may need to bifurcate the calls rather than just `send` GETs and POSTs identically. 🤔 (@nilesmc , thoughts?)

Yeah, I found the docs a bit confusing, but I've tested this against an actual API (the Google Address Validation API, to be specific) and it works. So maybe the docs are wrong / misleading?

**Comment by nilesmc** [comment]
> Unfortunately [the faraday docs](https://lostisland.github.io/faraday/#/getting-started/quick-start?id=get-head-delete-trace) imply POST requests send data as the _second_ argument (the third is for headers), so we may need to bifurcate the calls rather than just `send` GETs and POSTs identically. 🤔 (@nilesmc , thoughts?)

It's a little confusing as we are using send for the request type.

*  First is request method type
*  Second arg is actually the first arg (url). 
*  Third is body or param. 
*  Fourth could be headers.

Since we are all ready setting the url above, we dont need to reset it. Nil would actually be the place where we would set the url (but we dont need to).

`@response = connection.send(type, nil, payload)`

I am pretty sure this is correct.

File: lib/ts/client/generic_base_client.rb [hunk 2]
```diff
@@ -35,8 +36,14 @@ def build_connection(url)
   # e.g. http://www.ft.com/ & get or http://os.teamshares.com/companies & post
   # which enables single call method and centralized approach to result handling
   def call(connection, type, payload: nil)
+    # The Faraday docs refer to using `get` and `post` directly.
```
Comment (CrookedGrin) [inline lib/ts/client/generic_base_client.rb:39]: @nilesmc @kdonovan Added some clarifying comments here for the next poor fool to come along

File: lib/ts/client/generic_base_client.rb [hunk 3]
```diff
@@ -35,8 +36,14 @@ def build_connection(url)
   # e.g. http://www.ft.com/ & get or http://os.teamshares.com/companies & post
   # which enables single call method and centralized approach to result handling
   def call(connection, type, payload: nil)
```
Comment (CrookedGrin) [inline lib/ts/client/generic_base_client.rb:38]: @nilesmc @kdonovan Added some clarifying comments here for the next poor fool to come along

**Review by kdonovan — APPROVED** [review]
@nilesmc @CrookedGrin thanks for the clarifications! Love Niles' idea of smoke testing, but that was my only concern. 🚗

**Comment by nilesmc** [comment]
Using this branch locally, I was able to close a Lead in Buyout, and trigger a webhook to OS, then a get to Buyout from os...

After the whole Rube-Goldberg machine did its thing, it created a company in OS: 

Screenshot 2023-10-02 at 11 13 17 AM

TLDR; We are good to merge!
